### ISSUE TITLE
NTLM authentication support

### ISSUE THREAD

**Issue by svennela** [spring-cloud/spring-cloud-config#237]
Hi,

Having issues with config server accessing git repo. customer is using Microsoft Team Foundation Server for their git repo. When config server trying connect and attempt to authenticate it requires NTLM authentication. Do we support .e. ntlm authentication ??

**Comment by spencergibb** [comment]
We support whatever authentication schemes [jgit](https://eclipse.org/jgit/) supports. I simple search reveals [this](https://git.eclipse.org/c/jgit/jgit.git/commit/?id=98dd6e6abdba75d05f03b5b073659efe53182dc6), so there may be hope.

**Comment by spencergibb** [comment]
We don't have any explicit support yet.

**Comment by lonnix** [comment]
Spencergibb, does your previous comment mean you're not allowing use of ntml authentication even if jgit has it? The link you provided is from 2013 and I believe ntlm is supported now according to that same link.

If ntlm is indeed supported, what spring Config value would need to be set to use ntlm authentication? Something like spring.cloud.config.server.git.auth=ntlm perhaps?

**Comment by spencergibb** [comment]
I'm saying if it is supported by jgit we could support it.

**Comment by dasavik** [comment]
Hello, I see this post is very old. We are in the same situation, client is using TFS 2015 for their git repo. Does jgit support NTLM authentication now?

**Comment by ryanjbaxter** [comment]
Since this issue is still open, I don't believe so.

**Comment by dasavik** [comment]
thanks @ryanjbaxter. Can you please provide some alternate references. SSH or any other kind of implementation for that supports NTLM authentication?

**Comment by ryanjbaxter** [comment]
I cant because I am not familiar with it, sorry

**Comment by sgangapa** [comment]
i'm also having similar need. I need to configure git repo with TFS server. Pls confirm if any efforts. Thanks.

**Comment by gmcouto** [comment]
Yeah, I'm trying to have a config server to clone from TFS also. But to no avail.

**Comment by sgangapa** [comment]
I have tried even with file system back-end (for testing)  but the none of the properties are getting loaded. I can't use git or svn as back-end as my VCS is TFS..

**Comment by gmcouto** [comment]
Is there a reason for ```HttpClientConfigurableHttpConnectionFactory``` to use```HttpClientConnection``` as default connection instead of ```JDKHttpConnection```?

JDKHttpConnection support NTLM if you are on windows, using current user credentials. And HttpClientConnection has an issue with multiple WWW-Authenticate header.

I defined an empty ```public class SpringJDKConnectionFactory extends JDKHttpConnectionFactory implements ConfigurableHttpConnectionFactory``` and added as a bean. Circumventing all issues caused by ```HttpClientConnection```
